Serialiser_Engine/Compute/Serialise/IDynamicObject.cs

```diff
@@ -0,0 +1,112 @@
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.Engine.Base;
+using BH.Engine.Reflection;
+using BH.oM.Base;
+using BH.oM.Base.Attributes;
+using MongoDB.Bson.IO;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+
+namespace BH.Engine.Serialiser
+{
+    public static partial class Compute
+    {
+
+        /*******************************************/
+        /**** Private Methods                   ****/
+        /*******************************************/
+        
+        private static void SerialiseDynamicObject(this IDynamicObject value, BsonDocumentWriter writer, Type targetType)
+        {
+            if (value == null)
+            {
+                writer.WriteNull();
+                return;
+            }
+
+            writer.WriteStartDocument();
+
+            writer.WriteName("_t");
+            writer.WriteString(value.GetType().FullName);
+
+            if (value is IDynamicPropertyProvider)
+            {
+                foreach (string prop in value.PropertyNames())
+                {
+                    object result;
+                    if (BH.Engine.Base.Compute.TryRunExtensionMethod(value, "GetProperty", new object[] { prop }, out result))
+                    {
+                        writer.WriteName(prop);
+                        ISerialise(result, writer, typeof(object));
+                    }
+                    else
+                    {
+                        Base.Compute.RecordWarning($"Failed to collect property {prop} when serialising {value.GetType().Name}. This property will be ignored.");
+                    }
+                } 
+            }
+            else
+            {
+                foreach (PropertyInfo prop in value.GetType().GetProperties())
+                {
+                    if (prop.GetCustomAttribute<DynamicPropertyAttribute>() != null && typeof(IDictionary).IsAssignableFrom(prop.PropertyType))
+                    {
+                        Type[] genericArguments = prop.PropertyType.GenericTypeArguments;
+                        if (genericArguments.Length != 2 || !prop.PropertyType.GenericTypeArguments.First().IsEnum)
+                        {
+                            writer.WriteName(prop.Name);
+                            ISerialise(prop.GetValue(value), writer, prop.PropertyType);
+                            BH.Engine.Base.Compute.RecordWarning($"A dynamic property container of type {prop.PropertyType.ToString()} is not supported. It will be serialised as a regular property.");
+                        }
+                        else
+                        {
+                            Type valueType = genericArguments[1];
+                            IDictionary dic = prop.GetValue(value) as IDictionary;
+                            foreach (Enum key in dic.Keys.OfType<Enum>())
```

Comment:
Detail that has 0 impact on functionality, but just could make the JSON look a bit better
```suggestion
                            foreach (Enum key in dic.Keys.OfType<Enum>().OrderBy(x => x))
```

A bit of a trade-off, as this is an additional sorting done for every dynamic object, so would slow it down ever so slightly, but given it effectively is sorting a short collection of integers, I presume it will be quick enough.

Not saying we have to do this, but might be worth in?